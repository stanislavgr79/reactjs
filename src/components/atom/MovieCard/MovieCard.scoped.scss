.movie-card {
    padding-bottom: 20px;
}

.movie-image {
    max-inline-size: fit-content;
    padding-bottom: 10px;
}

.movie-props {
    .movie-info {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        color: #bbbbbb;
        padding-bottom: 5px;
        font-size: 18px;
        .movie-releaseDate {
            border-radius: 5px;
            font-size: 15px;
            border: 0.5px solid #bbbbbb;
            padding-left: 7.9px;
            padding-right: 7.9px;
            height: 25px;
            text-align: center;
            width: 50px;
        }
    }
    .movie-genre {
        font-size: 16px;
        color: #555555;
        padding-bottom: 5px;
    }
}

@media screen and (max-width: 1199px) {
    .movie-props .movie-info .movie-releaseDate {
        margin-right: 2%;
    }
}