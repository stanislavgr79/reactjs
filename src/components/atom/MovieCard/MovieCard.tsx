import React from 'react';
import './MovieCard.scoped.scss';

interface IProps {
  id: number | undefined;
  title: string;
  movieUrl?: string;
  releaseDate: string;
  genres: string[] | { value: string, label: string }[];
  posterPath: string | undefined;
  changeSelectedMovieId: (e: number) => void;
}

export default function MovieCard({
  id = 0,
  title = 'missing_title',
  releaseDate = 'missing_releaseDate',
  genres = ['missing_genre'],
  posterPath = 'missing_image',
  changeSelectedMovieId,
}: IProps): JSX.Element {
  const year = releaseDate.split('-')[0];
  const KEY = 'value';
  const genreStr =
    typeof genres[0] === 'object'
      ? genres
          .map(function (item: any) {
            return item[KEY];
          })
          .join(', ')
      : genres.join(', ');

  return (
    <>
      <div className="movie-card">
        <div className="movie-image">
          <a href="#">
            <img
              src={posterPath}
              className="img-fluid"
              alt={title}
              onClick={() => changeSelectedMovieId(id)}
            />
          </a>
        </div>
        <div className="movie-props">
          <div className="movie-info">
            <div className="movie-title">{title}</div>
            <div className="movie-releaseDate">{year}</div>
          </div>
          <div className="movie-genre">{genreStr}</div>
        </div>
      </div>
    </>
  );
}
