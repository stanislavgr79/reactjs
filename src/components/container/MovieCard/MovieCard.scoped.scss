.movie-card {
    margin-left: 10%;
    padding-bottom: 20px;
}

.movie-image {
    max-inline-size: fit-content;
    padding-bottom: 10px;
}

.movie-props {
    font-size: 18px;
    .movie-info {
        display: flex;
        flex-direction: row;
        justify-content: space-between;
        color: #bbbbbb;
        padding-bottom: 5px;
        .movie-releaseDate {
            border-radius: 5px;
            font-size: 15px;
            border: 0.5px solid #bbbbbb;
            width: 70px;
            height: 25px;
            margin-right: 18%;
            text-align: center;
            width: 50px;
        }
    }
    .movie-genre {
        color: #555555;
        padding-bottom: 5px;
    }
}

@media screen and (max-width: 1199px) {
    .movie-props .movie-info .movie-releaseDate {
        margin-right: 2%;
    }
}