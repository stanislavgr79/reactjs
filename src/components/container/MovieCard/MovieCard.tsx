import React from 'react';
import './MovieCard.scoped.scss';
import images from '../../../helpers/images';

interface MovieCardProps {
  title: string;
  releaseDate: string;
  movieUrl: string;
  genre: string;
  image: string;
  overview?: string;
  runtime?: number | string;
}

export default function MovieCard({
  title = 'missing_title',
  releaseDate = 'missing_releaseDate',
  movieUrl = 'missing_movieUrl',
  genre = 'missing_genre',
  image = 'missing_image',
}: MovieCardProps) {
  const imageRef = images.find((obj: any) => obj.name === image);
  const year = releaseDate.split('.')[2];
  return (
    <>
      <div className="container-md movie-card">
        <div className="movie-image">
          <a href={movieUrl}>
            <img src={imageRef ? imageRef?.value : ''} className="img-fluid" alt={imageRef?.name} />
          </a>
        </div>
        <div className="movie-props">
          <div className="movie-info">
            <div className="movie-title">{title}</div>
            <div className="movie-releaseDate">{year}</div>
          </div>
          <div className="movie-genre">{genre}</div>
        </div>
      </div>
    </>
  );
}
