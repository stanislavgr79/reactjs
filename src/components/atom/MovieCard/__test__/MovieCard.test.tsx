import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import MovieCard from '../MovieCard';

describe('MovieCard', () => {
  const movie = {
    id: 12345,
    title: 'film title',
    release_date: '2020-01-02',
    poster_path: '',
    genres: ['crime'],
    overview: 'overview',
    runtime: 66,
    vote_average: 5.5,
    tagline: '0',
  };

  test('render', () => {
    const tree = render(
      <BrowserRouter>
        <MovieCard movie={movie} />,
      </BrowserRouter>,
    );

    expect(tree.baseElement).toMatchSnapshot();
  });
});
