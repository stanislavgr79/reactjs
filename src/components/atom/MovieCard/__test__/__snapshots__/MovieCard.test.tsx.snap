// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard render 1`] = `
<body>
  <div>
    <div
      class="movie-card"
    >
      <div
        class="movie-image"
      >
        <a
          href="/movies/12345"
        >
          <img
            alt="film title"
            class="img-fluid"
            src=""
          />
        </a>
      </div>
      <div
        class="movie-props"
      >
        <div
          class="movie-info"
        >
          <div
            class="movie-title"
          >
            film title
          </div>
          <div
            class="movie-releaseDate"
          >
            2020
          </div>
        </div>
        <div
          class="movie-genre"
        >
          crime
        </div>
      </div>
    </div>
    ,
  </div>
</body>
`;
